engines/python/setup/djl_python/neuron_utils/model_loader.py

```diff
@@ -808,3 +808,115 @@ def partition(self, save_path: str, **kwargs):
+        self.model_config.generation_config = self.generation_config
+        self.neuron_config = self.generate_neuron_config()
+
+    def generate_generation_config(self):
+        generation_config = NeuronGenerationConfig(
+            max_length=self.config.max_model_len,
+            do_sample=True,
+            per_batch_line=True,
+            top_k=[1] * self.config.max_rolling_batch_size,
+            top_p=[1] * self.config.max_rolling_batch_size,
+            temperature=[1] * self.config.max_rolling_batch_size,
+            dynamic=True,
+            global_top_k=256,
+            deterministic=True,
+        )
+        return generation_config
+
+    def generate_neuron_config(self):
+        continuous_batching_config = None
+        if self.config.max_rolling_batch_size > 1:
+            continuous_batching_config = ContinuousBatchingConfig(
+                batch_size_for_shared_caches=self.config.max_rolling_batch_size)
+
+        on_dev_sampling_config = None
+        if self.config.on_device_generation:
+            on_dev_sampling_config = copy.deepcopy(self.model_config.generation_config)
+
+        quant_config = None
+        if self.config.neuron_quant:
+            quant_dtype = 's8'
+            dequant_dtype = 'bf16'
+            logging.info(f'Quantization is enabled. '
+                         f'quant_dtype: {quant_dtype}, '
+                         f'dequant_dtype: {dequant_dtype}')
+            quant_config = QuantizationConfig(
+                quant_dtype=quant_dtype,
+                dequant_dtype=dequant_dtype
+            )
+
+        weight_tiling = True
+        qkv_tiling = True
+        fuse_mlp = False
+        mlp_out_weight_transpose = False
+
+        if self.config.multi_node:
+            weight_tiling = False
+            qkv_tiling = False
+            fuse_mlp = True
+            mlp_out_weight_transpose = True
+
+        if self.config.speculative_draft_model:
+            weight_tiling = False
+            qkv_tiling = False
+            fuse_mlp = False
+            mlp_out_weight_transpose = False
+
+        neuron_config = NeuronConfig(
+            continuous_batching=continuous_batching_config,
+            attention_layout="BSH",
+            on_device_embedding=self.config.on_device_embedding,
+            on_device_generation=on_dev_sampling_config,
+            collectives_layout="BSH",
+            quant=quant_config,
+            fuse_qkv=True,
+            compilation_worker_count=self.config.compilation_worker_count,
+            sequence_parallel_norm=self.config.sequence_parallel,
+            sequence_parallel_norm_threshold=1,
+            shard_over_sequence=self.config.shard_over_sequence,
+            weight_tiling=weight_tiling,
+            qkv_tiling=qkv_tiling,
+            fuse_mlp=fuse_mlp,
+            mlp_out_weight_transpose=mlp_out_weight_transpose,
+        )
+        logging.info(f'Neuron config: {neuron_config}')
```

Comment:
Move to .debug statements, for nested objects you can change the __str__ or something similar so that this can be done in a one liner